src/configuration/configuration-base.ts

```diff
@@ -190,8 +196,22 @@ export default class Configuration {
         option.source = source;
     }
 
+    protected _addOverriddenOptionIfRequired (value1: OptionValue, value2: OptionValue, source: OptionSource, optionName: string): void {
```

Comment:
`_addOverridenOptionIfNecessary` is more suitable function name.

Reply:
ok